src/services/organizeImports.ts

```diff
@@ -322,96 +322,105 @@ function coalesceImportsWorker(importGroup: readonly ImportDeclaration[], compar
         return importGroup;
     }
 
-    const { importWithoutClause, typeOnlyImports, regularImports } = getCategorizedImports(importGroup);
-    const coalescedImports: ImportDeclaration[] = [];
-
-    if (importWithoutClause) {
-        coalescedImports.push(importWithoutClause);
-    }
-
-    for (const group of [regularImports, typeOnlyImports]) {
-        const isTypeOnly = group === typeOnlyImports;
-        const { defaultImports, namespaceImports, namedImports } = group;
-        // Normally, we don't combine default and namespace imports, but it would be silly to
-        // produce two import declarations in this special case.
-        if (!isTypeOnly && defaultImports.length === 1 && namespaceImports.length === 1 && namedImports.length === 0) {
-            // Add the namespace import to the existing default ImportDeclaration.
-            const defaultImport = defaultImports[0];
-            coalescedImports.push(
-                updateImportDeclarationAndClause(defaultImport, defaultImport.importClause.name, namespaceImports[0].importClause.namedBindings),
-            );
-
-            continue;
+    const importGroupsByAttributes = groupBy(importGroup, decl => {
+        if (decl.attributes) {
+            return decl.attributes.getFullText();

```

Comment:
Saw some discussion on Teams about this. As you were concerned about, this is sensitive to attribute order, spacing, comments, quote preference, etc. The grammar is simple enough that I think instead of doing something like emitting a normalized version of the text (which would still be sensitive to attribute order), you could pretty easily write something like

```ts
function getImportAttributesId(attributes: ImportAttributes) {
  const keyValuePairs = new Map<string, string>;
  // - iterate over attribute elements
  // - value must be a string literal expression if valid
  // - if invalid, insert something unique?
  let id = "";
  const sorted = sort(
    arrayFrom(keyValuePairs.entries()),
    ([keyA], [keyB]) => compareStringsCaseSensitive(keyA, keyB)
  );
  for (const [key, value] of sorted) {
    // value.length included in case `key` contains `:`,
    // a bit over the top but technically something like
    // this is needed
    id += `${key}:${value}:${value.length}|`;
  }
  return id;
}